src/openms/include/OpenMS/TRANSFORMATIONS/RAW2PEAK/PeakPickerHiRes.h

```diff
@@ -447,20 +449,34 @@ namespace OpenMS
 
       Size progress = 0;
       startProgress(0, input.size() + input.getChromatograms().size(), "picking peaks");
-      for (Size scan_idx = 0; scan_idx != input.size(); ++scan_idx)
+
+      if (!input.getNrSpectra() == 0)
```

Comment:
ehmm.. operator precedence.
'!' will be eval'd before == ...
Do you mean 'if (input.getNrSpectra() > 0)'?
It's actually doing the right thing (I think) but there is bool conversion going on if I'm not totally off track.